apps/crn-server/test/data-providers/contentful/users.data-provider.test.ts

```diff
@@ -1,34 +1,519 @@
+              },
+            ],
+          },
+        }),
+      });
+
+      const result = await userDataProvider.fetchById('123');
+      expect(result!.labs).toHaveLength(1);
+      expect(result!.labs[0]!.name).toEqual('My lab');
+    });
+
+    test('should remove empty values from expertise tags and questions', async () => {
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+        users: getContentfulGraphqlUser({
+          questions: [null, '', 'Why?'],
+          expertiseAndResourceTags: ['Research', null, ''],
+        }),
+      });
+      const result = await userDataProvider.fetchById('123');
+      expect(result!.questions).toEqual(['Why?']);
+      expect(result!.expertiseAndResourceTags).toEqual(['Research']);
+    });
+
+    test('should tag alumni users', async () => {
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+        users: getContentfulGraphqlUser({
+          alumniSinceDate: '2020-01-01T12:00:00.000Z',
+        }),
+      });
+
+      const result = await userDataProvider.fetchById('123');
+
+      expect(result!._tags).toEqual(['Alumni Member']);
+    });
+
+    test('should tag non-alumni users', async () => {
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+        users: getContentfulGraphqlUser({
+          alumniSinceDate: null,
+        }),
+      });
+
+      const result = await userDataProvider.fetchById('123');
+
+      expect(result!._tags).toEqual(['CRN Member']);
     });
   });
 
   describe('Fetch', () => {
-    test('should receive a fixed user list', async () => {
-      const result = await userDataProvider.fetch();
+    test('should receive a user list', async () => {
+      const contentfulGraphqlClientMockServer =
+        getContentfulGraphqlClientMockServer(getContentfulGraphql());
+
+      const userDataProviderWithMockServer: UserDataProvider =
+        new UserContentfulDataProvider(
+          contentfulGraphqlClientMockServer,
+          contentfulRestClientMock,
+        );
+      const result = await userDataProviderWithMockServer.fetch({});
+      const expectation = getUserDataObject();
+      // TODO: attach ORCID works to users
+      expectation.orcidWorks = [];
+      // TODO: team proposal
+      expectation.teams[0]!.proposal = undefined;
+      expect(result.total).toEqual(1);
+      expect(result.items).toEqual([expectation]);
+    });
+
+    test('should return an empty response if there is no result', async () => {
+      contentfulGraphqlClientMock.request.mockResolvedValueOnce({
+        usersCollection: null,
+      });
+      const result = await userDataProvider.fetch({});
       expect(result).toEqual({
-        items: [expect.any(Object)],
-        total: 1,
+        total: 0,
+        items: [],
       });
     });
-  });
 
-  describe('Create', () => {
-    test('should throw an error', async () => {
-      await expect(userDataProvider.create()).rejects.toThrow();
+    describe('query parameters', () => {
```

Comment:
So the types as defined [here](https://github.com/yldio/asap-hub/blob/01c6029095578374558185167fd3dacc29e27074/packages/model/src/user.ts#L222-L248) should specify exactly one filter per query for linked filters.

My assumption is that if we added a unit test to cover this case also then it would fail type-checking?

Reply:
yeah no - i hear you, you would need to cast as unknown first and i think TS checks are good enough, I guess what I meant though was that i think you can still do a text search with these filters, whereas that functionality has actually been removed no?

Reply:
nvm i see you've address it already